Framework/Algorithms/src/StripPeaks.cpp

```diff
@@ -175,21 +175,21 @@ StripPeaks::removePeaks(API::MatrixWorkspace_const_sptr input,
   // progress 0.2 to 0.3 in this loop
   double prg = 0.2;
   for (size_t k = 0; k < hists; ++k) {
-    outputWS->dataX(k) = input->readX(k);
-    outputWS->dataY(k) = input->readY(k);
-    outputWS->dataE(k) = input->readE(k);
+    outputWS->setHistogram(k, input->histogram(k));
     prg += (0.1 / static_cast<double>(hists));
     progress(prg);
   }
 
-  const bool isHistogramData = outputWS->isHistogramData();
   // progress from 0.3 to 1.0 here
   prg = 0.3;
   // Loop over the list of peaks
   for (size_t i = 0; i < peakslist->rowCount(); ++i) {
-    // Get references to the data
-    const MantidVec &X = outputWS->readX(peakslist->getRef<int>("spectrum", i));
-    MantidVec &Y = outputWS->dataY(peakslist->getRef<int>("spectrum", i));
+    // Get references to the data - X can be const
+    // but Y needs to be mutable
+    auto &X = outputWS->x(peakslist->getRef<int>("spectrum", i));
+    auto &Y = outputWS->mutableY(peakslist->getRef<int>("spectrum", i));
+	// Also create point data for later
+	auto &pointData = outputWS->points(i);
```

Comment:
Probably this will not compile. `points()` returns by value, not by reference (the type`Point` is wrapping a pointer, so return by value is reasonably cheap).